### DISCUSSION TITLE
Why is there a time difference between this app and mobile authenticator?

### DISCUSSION THREAD

**Discussion by ghost** [FireCubeStudios/Protecc#66] [Q&A]
When I add the TOTP secret of 2FA from a website like Twitter both to Protecc and Microsoft Authenticator on my Android, I notice there is 2 or 3 seconds difference between the times of the codes.
each of them has 30 seconds but for example on the phone it's 3 seconds ahead of Protecc on my Windows PC.

both use the same secret so there isn't any issue with that, I'm curious to know what causes this time difference?

what I'm most afraid of is whether this time difference will get bigger and bigger over time and reach a point where the codes generated will become invalid. is that a possibility?

**Comment by FireCubeStudios** [comment]
I don't know why this is happening. Possibly time server mismatch in windows I don't think this will be a major issue

**Reply by ghost** [reply]
Do you know if having a de-synced time at the time of creation of the TOTP secret on a website, can have any effect on the codes that apps like Protecc produce later on?

**Reply by FireCubeStudios** [reply]
If desync isn't big then it's fine. I could have added time customisation option in protecc but even that was too complex for me to implement but maybe in the refactor I can include it

**Reply by ghost** [reply]
Sorry but what's refactor?
